app/services/settings/authentication/upsert.rb

```diff
@@ -11,7 +11,10 @@ def self.call(settings)
       end
 
       def self.update_enabled_providers(value)
-        return if value.blank?
+        if value.blank?
```

Comment:
If we get rid of the guard clause completely, `enabled_providers` will end up as an empty array because of the `filter_map` call (`"".split(",")` returns `[]` which then becomes the result of `filter_map`). This way we would also still perform the `email_login_disabled_with_one_or_less_auth_providers` check, which seems like a good idea to me.

I made the following change locally and all of the newly added specs you added still pass:

```diff
diff --git a/app/services/settings/authentication/upsert.rb b/app/services/settings/authentication/upsert.rb
index 4cb8002c4..87646b967 100644
--- a/app/services/settings/authentication/upsert.rb
+++ b/app/services/settings/authentication/upsert.rb
@@ -11,11 +11,6 @@ def self.call(settings)
       end

       def self.update_enabled_providers(value)
-        if value.blank?
-          Settings::Authentication.providers = []
-          return
-        end
-
         enabled_providers = value.split(",").filter_map do |entry|
           entry unless invalid_provider_entry(entry)
         end
```

I don't think `auth_providers_to_enable` will ever be `nil` but if we are worried it could be you could always go for `value.to_s.split(",").